matminer/featurizers/base.py

```diff
@@ -151,6 +155,84 @@ def set_chunksize(self, chunksize):
     def chunksize(self):
         return self._chunksize if hasattr(self, '_chunksize') else None
 
+    def precheck_dataframe(self, df, col, return_frac=True, inplace=False) \
+            -> [float, pd.DataFrame]:
+        """
+        Precheck an entire dataframe. Subclasses wanting to use precheck
+        functinoality should not override this method, they should override
+        precheck (unless the entire df determines whether single entries pass
+        or fail a precheck).
+
+        Prechecking should be a quick and useful way to check that for a
+        particular dataframe (set of featurizer inputs), the featurizer is:
+
+            1. in scope, and/or...
+            2. robust to errors and/or...
+            3. any other reason you would not practically want to use this
+                featurizer in on this dataframe.
+
+        By prechecking before featurizing, you can avoid applying featurizers
+        to data that will ultimately fail, return unreliable numbers, or
+        are out of scope. Prechecking is also a good time to throw/observe
+        warnings (such as long runtime warnings!).
+
+        Args:
+            df (pd.DataFrame): A dataframe
+            col (str): The column in df specifying the input to the featurizer.
+            return_frac (bool): If True, returns the fraction of entries
+                passing the precheck (e.g., 0.5). Else, returns a dataframe.
+            inplace (bool); Only relevant if return_frac=False. If inplace=True,
+                the input dataframe is modified in memory with a boolean column
+                for precheck. Otherwise, a new df with this column is returned.
+
+        Returns:
+            (bool, pd.DataFrame): If return_frac=True, returns the fraction of
+                entries passing the precheck. Else, returns the dataframe with
+                an extra boolean column added for the precheck.
+
+        """
+        prechecks = [self.precheck(e) for e in df[col]]
+
+        if return_frac:
+            return np.sum(prechecks) / len(prechecks)
+        else:
+            precheck_col = "{} precheck pass".format(self.__class__.__name__)
+            if inplace:
+                df[precheck_col] = prechecks
+            else:
+                res = pd.DataFrame({precheck_col: prechecks})
+                df = pd.concat([df, res], axis=1)
+            return df
+
+    def precheck(self, x) -> bool:
```

Comment:
As above, should we change this to `*x` to allow for >1 inputs? 

PS. I like the use of typing! Yay for dropping 2.x support!

Reply:
Sure will do